llrt_core/src/modules/http/request.rs

```diff
@@ -207,18 +220,33 @@ fn assign_request<'js>(request: &mut Request<'js>, ctx: Ctx<'js>, obj: &Object<'
                 ));
             }
 
-            request.body = if let Some(blob) = get_class::<Blob>(&body)? {
-                let blob = blob.borrow();
-                Some(TypedArray::<u8>::new(ctx.clone(), blob.get_bytes())?.into_value())
-            } else {
-                Some(body)
+            match get_class::<Blob>(&body)? {
+                Some(blob) => {
+                    let blob = blob.borrow();
+                    request.body =
+                        Some(TypedArray::<u8>::new(ctx.clone(), blob.get_bytes())?.into_value());
+                    request.content_type = match blob.mime_type().len() {
+                        0 => None,
+                        _ => Some(blob.mime_type()),
+                    };
+                },
+                None => {
+                    request.body = Some(body);
+                    request.content_type = None;
+                },
```

Comment:
This can be implemented like this instead:
```rust
if let Some(blob) = get_class::<Blob>(&body)? {
    let blob = blob.borrow();
    request.body = Some(TypedArray::<u8>::new(ctx.clone(), blob.get_bytes())?.into_value());
    request.content_type = if blob.mime_type().is_empty() {
        None
    } else {
        Some(blob.mime_type())
    };
} else {
    request.body = Some(body);
    request.content_type = None;
}
```

Reply:
I have decided to refer to the content-type of Headers directly in Blob(), so this support is no longer necessary.